src/Validation/Validator.php

```diff
@@ -2033,6 +2034,45 @@ public function email(
         ]);
     }
 
+    /**
+     * Add a backed enum validation rule to a field.
+     *
+     * @param string $field The field you want to apply the rule to.
+     * @param class-string<\BackedEnum> $enumClassName The valid backed enum class name.
+     * @param string|null $message The error message when the rule fails.
+     * @param \Closure|string|null $when Either 'create' or 'update' or a Closure that returns
+     *   true when the validation rule should be applied.
+     * @return $this
+     * @see \Cake\Validation\Validation::enum()
+     * @since 5.1.0
+     */
+    public function enum(
+        string $field,
+        string $enumClassName,
+        ?string $message = null,
+        Closure|string|null $when = null
+    ) {
+        if (!in_array(BackedEnum::class, (array)class_implements($enumClassName), true)) {
+            throw new InvalidArgumentException(
+                'The `$enumClassName` argument must be the classname of a valid backed enum.'
+            );
+        }
+
+        if ($message === null) {
+            if (!$this->_useI18n) {
+                $message = sprintf('The provided value must be a `\%s` enum instance or value', $enumClassName);
+            } else {
+                $message = __d('cake', 'The provided value must be a `\{0}` enum instance or value', $enumClassName);
+            }
```

Comment:
While these are technically correct validation messages, I'm not sure they will be coherent to end users. What if we had an error message similar to `inList`?

```suggestion
            $cases = array_map(fn ($case) => $case->value, $enumClassName::cases());
            $caseOptions = implode(', ', $cases);
            if (!$this->_useI18n) {
                $message = sprintf('The provided value must be one of %s', $caseOptions);
            } else {
                $message = __d('cake', 'The provided value must be one of {0}', $caseOptions);
            }
```

Reply:
Makes sense, I was thinking with developers in mind, but clearly these messages are primarily user facing.